backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/ContextUtils.kt

```diff
@@ -399,6 +399,8 @@ internal class Llvm(val context: Context, val llvmModule: LLVMModuleRef) {
         context.config.resolvedLibraries
                 .getFullList(TopologicalLibraryOrder)
                 .filter { (!it.isDefault && !context.config.purgeUserLibs) || imports.bitcodeIsUsed(it) }
+                // TODO: the filter above is incorrect when compiling to multiple LLVM modules.
```

Comment:
So what you suggest to do with problem mentioned in TODO?

Reply:
It mostly depends on how exactly compiling to multiple LLVM modules will be used.
The PR doesn't define this, so it doesn't address the problem.

For example, for build cache we may decide to include the bitcode here unconditionally, i.e. remove the filter above (since it is much like DCE, and compiling to cache is generally supposed to disable DCE).

More general solution may eventually involve getting rid of the prepared bitcode in interop libraries or privately linking it into each LLVM module that uses this bitcode.

Reply:
Also we could consider finally dropping the concept of default libraries that are supposed to be DCEd (as it brings a lot of issues and complicates the implementation of many components), requiring configuring them explicitly (e.g. with `-l Foundation` in CLI or dependencies in Gradle).